### PR TITLE
[oracle_weblogic] Add Integration Package with ThreadPool Data Stream

### PR DESCRIPTION
**Pull request by kush-elastic**
- Enhancement

## What does this PR do?

- Generated the skeleton of Oracle WebLogic integration package.
- Added 1 data stream (ThreadPool).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.
- Added system test cases for the data stream.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/oracle_weblogic directory.
- Run the following command to run tests.
`elastic-package test`

## Related issues

- Relates #2823
- Relates #2910
- Relates #3066
- Relates #3067
- Relates #3068
- Relates #3069
- Relates #3121

## Screenshots

oracle_weblogic-threadpool-application

### CODE REVIEW

File: packages/oracle_weblogic/_dev/build/docs/README.md [hunk 1]
```diff
@@ -6,6 +6,25 @@ The Oracle WebLogic Integration is used to fetch observability data from [Oracle
 
 This integration has been tested against `Oracle WebLogic v12.2.1.3`.
 
+## Requirements
+
+In order to ingest data from Oracle WebLogic:
+- You must know the host for Oracle WebLogic application, add that host while configuring the integration package.
+- Add default path for jolokia.
```
Comment (yug-elastic) [inline packages/oracle_weblogic/_dev/build/docs/README.md:13]: nit: `Jolokia`

File: packages/oracle_weblogic/_dev/build/docs/README.md [hunk 2]
```diff
@@ -6,6 +6,25 @@ The Oracle WebLogic Integration is used to fetch observability data from [Oracle
...
+- Configuring Jolokia for Weblogic
```
Comment (yug-elastic) [inline packages/oracle_weblogic/_dev/build/docs/README.md:14]: nit: `WebLogic`

File: packages/oracle_weblogic/data_stream/threadpool/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  - rename:
+      field: message
+      target_field: event.original
+      ignore_missing: true
+  - rename:
+      field: jolokia.metrics
+      target_field: oracle_weblogic
+      ignore_missing: true
+      ignore_failure: true
+  - remove:
```
Comment (yug-elastic) [inline packages/oracle_weblogic/data_stream/threadpool/elasticsearch/ingest_pipeline/default.yml:17]: Please put this `remove` processor at the end

**Review by yug-elastic — CHANGES_REQUESTED** [review]
Requested some changes

**Review by yug-elastic — APPROVED** [review]
LGTM!

File: packages/oracle_weblogic/kibana/lens/oracle_weblogic-c31dabf0-c132-11ec-aaae-35e5cf8554e6.json [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                "query": ""
+            },
+            "visualization": {
+                "accessor": "4529912a-42b0-4057-8444-ed67f9cbbd8b",
+                "colorMode": "None",
+                "layerId": "17891fa4-cd52-4347-8ba4-158f2f99ceaf",
+                "layerType": "data"
+            }
+        },
+        "title": "Live Daemon and Non-Daemon Threads [Metrics Oracle WebLogic]",
```
Comment (agithomas) [inline packages/oracle_weblogic/kibana/lens/oracle_weblogic-c31dabf0-c132-11ec-aaae-35e5cf8554e6.json:83]: Please refer to the previous comment

File: packages/oracle_weblogic/data_stream/threadpool/fields/fields.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+              description: The threads that are being held by a request right now. These threads will either be declared as stuck after the configured timeout or will return to the pool before that. The self-tuning mechanism will backfill if necessary.
+            - name: standby
+              type: long
+              description: The number of threads in the standby pool. Threads that are not needed to handle the present work load are designated as standby and added to the standby pool. These threads are activated when more threads are needed.
+            - name: stuck
+              type: long
+              description: Number of stuck threads in the thread pool.
+            - name: total
+              type: long
+              description: Current number of live daemon and non-daemon threads.
```
[the remark this answers is not in the document]
Comment (agithomas) [reply packages/oracle_weblogic/data_stream/threadpool/fields/fields.yml]: You may need to update  below files

 the 
```
packages/oracle_weblogic/img/oracle_weblogic-threadpool.png
packages/oracle_weblogic/kibana/lens/oracle_weblogic-c31dabf0-c132-11ec-aaae-35e5cf8554e6.json
```

**Review by agithomas — APPROVED** [review]
Looks good to me

**Pull request merged**